src/txdb.cpp

```diff
@@ -44,10 +45,21 @@ uint256 CCoinsViewDB::GetBestBlock() const {
     return hashBestChain;
 }
 
+uint256 CCoinsViewDB::GetUptoBlock() const {
+    uint256 hashBestChain;
+    if (!db.Read(DB_BEST_BLOCK_UPTO, hashBestChain))
+        return uint256();
+    return hashBestChain;
+}
+
 bool CCoinsViewDB::BatchWrite(CCoinsMap &mapCoins, const uint256 &hashBlock) {
     CDBBatch batch(db);
     size_t count = 0;
     size_t changed = 0;
+    size_t batch_size = (size_t)GetArg("-dbbatchsize", nDefaultDbBatchSize) << 20;
```

Comment:
If we have memory usage that is some multiple of this, perhaps the argument should be in the form of the actual usage rather than the batch size?